ballerina-tests/tests/http_cache_config_annotation_test.bal

```diff
@@ -0,0 +1,303 @@
+        if (response is http:Response) {
+            response.setHeader(serviceHitCount, numberOfHitsNew.toString());
+            response.setHeader(proxyHitCount, numberOfProxyHitsNew.toString());
+            return response;
+        } else {
+            http:InternalServerError errorRes = {body : response.message()};
+            return errorRes;
+        }
+    }
+
+    resource function get statusResponse(http:Request req) returns http:Response|http:InternalServerError {
+        http:Response|error response = cacheBackendEP->forward("/statusResponseBE", req);
+        if (response is http:Response) {
+            return response;
+        } else {
+            http:InternalServerError errorRes = {body : response.message()};
+            return errorRes;
+        }
+    }
+}
+
+service / on new http:Listener(cacheAnnotationTestPort2) {
+
+    resource function default nocacheBE(http:Request req) returns @http:CacheConfig{noCache : true, maxAge : -1,
+    mustRevalidate : false} json {
+        noCacheHitCountNew += 1;
+        if (noCacheHitCountNew == 1) {
+            return nocachePayload1;
+        } else {
+            return nocachePayload2;
+        }
+    }
+
+    resource function default maxAgeBE(http:Request req) returns @http:CacheConfig{maxAge : 5, mustRevalidate : false} xml {
+        maxAgeHitCountNew += 1;
+        if (maxAgeHitCountNew == 1) {
+            return maxAgePayload1;
+        } else {
+            return maxAgePayload2;
+        }
+    }
+
+    resource function get mustRevalidateBE(http:Request req) returns @http:CacheConfig{maxAge : 5} string|byte[] {
+        numberOfHitsNew += 1;
+        if (numberOfHitsNew < 2) {
+            return mustRevalidatePayload1;
+        } else {
+            return mustRevalidatePayload2;
+        }
+    }
+
+    resource function get statusResponseBE(http:Request req) returns @http:CacheConfig{noCache : true, maxAge : -1,
+    mustRevalidate : false} http:Ok|http:InternalServerError {
+        statusHits += 1;
+        if (statusHits < 3) {
+            return ok;
+        } else {
+            return err;
+        }
+    }
+}
+
+@test:Config {}
+function testNoCacheCacheControlWithAnnotation() {
+    http:Response|error response = cacheClientEP->get("/noCache");
+    if (response is http:Response) {
+        test:assertEquals(response.statusCode, 200, msg = "Found unexpected output");
+        test:assertEquals(noCacheHitCountNew, 1);
+        test:assertTrue(response.hasHeader(LAST_MODIFIED));
+        assertHeaderValue(checkpanic response.getHeader(CACHE_CONTROL), "no-cache,public");
+        assertHeaderValue(checkpanic response.getHeader(ETAG), crypto:crc32b(nocachePayload1.toString().toBytes()));
+        assertHeaderValue(checkpanic response.getHeader(CONTENT_TYPE), APPLICATION_JSON);
+        assertJsonPayload(response.getJsonPayload(), nocachePayload1);
+    } else {
+        test:assertFail(msg = "Found unexpected output type: " + response.message());
```

Comment:
we can use `check` to leave out error but need to change the function returns as below. 

```ballerina
function testNoCacheCacheControlWithAnnotation() returns error? {
    http:Response response = check cacheClientEP->get("/noCache");
}
```